### PR TITLE
KIP 396 ListOffsets

### PR DESCRIPTION
**Pull request by mahajanadhitya**

### CODE REVIEW

File: kafka/integration_test.go
Comment (milindl) [inline kafka/integration_test.go:2715]: Now that assertions are available, use them.
```
assert := its.Assert()
....
....
assert.Nil(err, "CreateTopics should succeed")
```
and other methods on assert.
Everywhere, it's better readability and we will use this library going forward

**Review by milindl — CHANGES_REQUESTED** [review]
Some initial comments for Round 1. 
Mostly addresses the code, some comments on test, and example is not reviewed.
For the public API stuff and how OffsetSpec will be used, we can also discuss internally with everyone to arrive at a consensus on all the bindings at once.

File: kafka/adminapi.go
Comment (mahajanadhitya) [reply kafka/adminapi.go:2769]: I will further discuss it with emanuelle.

File: kafka/integration_test.go
Comment (milindl) [reply kafka/integration_test.go:2747]: Similar kind of language should be used everywhere, but this is a nit

File: kafka/adminapi.go
Comment (milindl) [inline kafka/adminapi.go:2779]: Move this up with other cTo... functions

File: kafka/adminoptions.go
Comment (milindl) [inline kafka/adminoptions.go:209]: Documentation comment above this missing

File: kafka/integration_test.go
Comment (milindl) [inline kafka/integration_test.go:2688]: Partition*

^ spelling

File: kafka/adminoptions.go
Comment (milindl) [inline kafka/adminoptions.go:181]: Missing comment above this

Comment (milindl) [reply kafka/adminoptions.go:181]: See other stuff around this for how to write the comment

**Review by milindl — CHANGES_REQUESTED** [review]
Example is missing from README.md of the examples/ directory, and also the .gitgnore (see how the other examples are added)

I also checked valgrind for memory leaks in the example. No issues

Mostly, great work, thanks @mahajanadhitya !

File: kafka/adminoptions.go
Comment (milindl) [inline kafka/adminoptions.go:218]:
```suggestion
// Valid for ListOffsets.
```

Comment (milindl) [reply kafka/adminoptions.go:218]: . It's valid only for ListOffsets

File: kafka/integration_test.go
Comment (milindl) [inline kafka/integration_test.go:2764]: The message should be changed too, not just the assert. 
Like
```suggestion
		assert.Equal(info.Error.Code(), ErrNoError, "Error code should be ErrNoError.")
```

Comment (milindl) [reply kafka/integration_test.go:2764]: At all three places.

File: kafka/adminoptions.go
Comment (milindl) [inline kafka/adminoptions.go:185]:
```suggestion
// Valid for ListOffsets.
```

**Review by milindl — COMMENTED** [review]
Minor comments on the addressal itself.
Almost ready to merge.

**Review by milindl — APPROVED** [review]
Approved, don't merge until librdkafka bundle is merged.
Good work @mahajanadhitya !

File: kafka/adminapi.go
Comment (emasab) [inline kafka/adminapi.go:1023]:
```suggestion
	ResultsInfos map[TopicPartition]ListOffsetsResultInfo
```

File: kafka/adminapi.go
Comment (emasab) [inline kafka/adminapi.go:1017]: Should be nil if -1, as in TopicPartition
```suggestion
	LeaderEpoch *int32
```

File: kafka/adminapi.go
Comment (emasab) [inline kafka/adminapi.go:1414]:
```suggestion
		resultInfo := ListOffsetsResultInfo{}
```

File: kafka/adminapi.go
Comment (emasab) [inline kafka/adminapi.go:1015]:
```suggestion
	Offset      kafka.Offset
```

File: kafka/adminapi.go
Comment (emasab) [inline kafka/adminapi.go:1420]:
```suggestion
		cLeaderEpoch := int32(C.rd_kafka_topic_partition_get_leader_epoch(cPartition))
		if cLeaderEpoch >= 0 {
			resultInfo.LeaderEpoch = &cLeaderEpoch
		}
```

File: kafka/adminoptions.go
Comment (emasab) [inline kafka/adminoptions.go:183]: As other enums:
```suggestion
	IsolationLevelReadUncommitted = IsolationLevel(C.RD_KAFKA_ISOLATION_LEVEL_READ_UNCOMMITTED)
```

File: kafka/adminoptions.go
Comment (emasab) [inline kafka/adminoptions.go:185]:
```suggestion
	IsolationLevelReadCommitted = IsolationLevel(C.RD_KAFKA_ISOLATION_LEVEL_READ_COMMITTED)
```

File: kafka/adminapi.go
Comment (emasab) [inline kafka/adminapi.go:1418]:
```suggestion
		Value.Offset = kafka.Offset(cPartition.offset)
```

**Review by emasab — CHANGES_REQUESTED** [review]
Great job! Some small changes

**Comment by milindl** [comment]
Addressed comments and merged master.

File: kafka/adminapi.go
Comment (milindl) [reply kafka/adminapi.go]: Just using Offset() here since the package is already kafka

File: kafka/adminapi.go
Comment (milindl) [reply kafka/adminapi.go]: Just 'Offset'

**Review by emasab — APPROVED** [review]
LGTM!

**Pull request merged**
